code/modules/clothing/head/_head.dm

```diff
@@ -27,4 +29,23 @@
 	..()
 	if(ismob(loc))
 		var/mob/M = loc
-		M.update_inv_head()
\ No newline at end of file
+		M.update_inv_head()
+
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	..()
+	if(thrownthing.target_zone && thrownthing.target_zone != BODY_ZONE_HEAD)
+		return
+	if(iscarbon(hit_atom))
+		var/mob/living/carbon/H = hit_atom
+		if(H.head)
+			var/obj/item/clothing/head/WH = H.head
+			if(!WH.snug_fit && !anti_tinfoil_maneuver)

```

Comment:
Pretty sure we're gonna need an istype here
```suggestion
			if(istype(WH) && !WH.snug_fit && !anti_tinfoil_maneuver)
```

Reply:
you need to type-check H.head before assigning it to WH, so at line 40 if(H.head && istype(H.head, /obj/item/clothing/head))

Reply:
Nah both spots will work fine. You can istype something after you typecast it. If the check is added right here, you still get the message that it bounces off. Adding it to line 40 skips this.